src/charm.py

```diff
@@ -95,19 +122,39 @@ def _on_install(self, event):
         else:
             self.unit.status = BlockedStatus("Missing token or org/repo path config")
 
-    def _on_upgrade_charm(self, event):
+    def _on_upgrade_charm(self, event: UpgradeCharmEvent) -> None:
+        """Handle the update of charm.
+
+        Args:
+            event (UpgradeCharmEvent): Event of charm upgrade.
+        """
         RunnerManager.install_deps()
 
-    def _on_config_changed(self, event):
-        self._ensure_event_timer("update-runner-bin", self.config["update-interval"])
-        self._ensure_event_timer("reconcile-runners", self.config["reconcile-interval"])
+    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
+        """Handle the configuration change.
+
+        Args:
+            event (ConfigChangedEvent): Event of configuration change.
+        """
+        try:
+            self._event_timer.ensure_event_timer(
+                "update-runner-bin", self.config["update-interval"]
+            )
+            self._event_timer.ensure_event_timer(
+                "reconcile-runners", self.config["reconcile-interval"]
+            )
+        except TimerEnableError as ex:
+            logger.exception("Failed to start the event timer")
```

Comment:
I think it would be nice here to log the original error message to gain some insight on what did actually happened when executing the command

Reply:
The `exception` method automatically logs the trace stack in it context.

For example, running the following script:
```
import logging

logger = logging.getLogger()


if __name__ == "__main__":
    try:
        int("two")
    except Exception as ex:
        logger.exception("Convert error")
```

Outputs:
```
Convert error
Traceback (most recent call last):
  File "/home/user/work/charm/github-runner-operator/test.py", line 8, in <module>
    int("two")
ValueError: invalid literal for int() with base 10: 'two'
```

`Logger.exception` is just `Logger.error` with the arg `exc_info` set to true. If `exc_info` is set to true it will print out the stack trace.

I only learn of this by reading this code base, since it was using `Logger.exception`.

Reply:
Didn't know that either, that is indeed quite handy

Reply:
Same here

Reply:
TIL, thanks @yhaliaw ! :)